synapse/lib/stormlib/easyperm.py

```diff
@@ -0,0 +1,93 @@
+                      {'name': 'scope', 'type': 'str', 'desc': 'The scope, either "users" or "roles".'},
+                      {'name': 'iden', 'type': 'str', 'desc': 'The user/role iden depending on scope.'},
+                      {'name': 'level', 'type': 'int', 'desc': 'The permission level number, or None to remove the permission.'},
+                  ),
+                  'returns': {'type': 'dict', 'desc': 'Dictionary with the updated easy perm structure.'}}},
+        {'name': 'allowed', 'desc': 'Check if the current user has a permission level in an easy perm dictionary.',
+         'type': {'type': 'function', '_funcname': '_allowedEasyPerm',
+                  'args': (
+                      {'name': 'edef', 'type': 'dict', 'desc': 'The easy perm dictionary to check.'},
+                      {'name': 'level', 'type': 'str', 'desc': 'The required permission level number.'},
+                  ),
+                  'returns': {'type': 'boolean',
+                              'desc': 'True if the user meets the requirement, false otherwise.', }}},
+        {'name': 'confirm', 'desc': 'Require that the current user has a permission level in an easy perm dictionary.',
+         'type': {'type': 'function', '_funcname': '_confirmEasyPerm',
+                  'args': (
+                      {'name': 'edef', 'type': 'dict', 'desc': 'The easy perm dictionary to check.'},
+                      {'name': 'level', 'type': 'str', 'desc': 'The required permission level number.'},
+                  ),
+                  'returns': {'type': 'null'}}},
+    )
+    _storm_lib_path = ('auth', 'easyperm')
+
+    def getObjLocals(self):
+        return {
+            'set': self._setEasyPerm,
+            'init': self._initEasyPerm,
+            'allowed': self._allowedEasyPerm,
+            'confirm': self._confirmEasyPerm,
+            'level': {
+                'deny': s_cell.PERM_DENY,
+                'read': s_cell.PERM_READ,
+                'edit': s_cell.PERM_EDIT,
+                'admin': s_cell.PERM_ADMIN,
+            }
+        }
+
+    async def _setEasyPerm(self, edef, scope, iden, level):
+        edef = await s_stormtypes.toprim(edef)
+        scope = await s_stormtypes.tostr(scope)
+        iden = await s_stormtypes.tostr(iden)
+        level = await s_stormtypes.toint(level, noneok=True)
+
+        await self.runt.snap.core._setEasyPerm(edef, scope, iden, level)
+        return edef
+
+    async def _initEasyPerm(self, edef=None):
+        edef = await s_stormtypes.toprim(edef)
+        if edef is None:
+            edef = {}
+
+        self.runt.snap.core._initEasyPerm(edef)
+
+        if self.runt.user is not None:
+            await self.runt.snap.core._setEasyPerm(edef, 'users', self.runt.user.iden, s_cell.PERM_ADMIN)
+        return edef
+
+    async def _allowedEasyPerm(self, edef, level):
+        if self.runt.user:
+            return self.runt.snap.core._hasEasyPerm(edef, self.runt.user, level)
+        return True
```

Comment:
when would we not have a runtime user?

Reply:
Will remove, for some reason I was thinking it was possible but I don't think it is.